includes/features/onboarding/class-wc-admin-onboarding.php

```diff
@@ -26,6 +26,65 @@ public static function get_instance() {
 		}
 		return self::$instance;
 	}
+
+	/**
+	 * Hook into WooCommerce.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_components_settings', array( $this, 'component_settings' ), 20 ); // Run after WC_Admin_Loader.
+		add_filter( 'admin_body_class', array( $this, 'add_admin_body_classes' ) );
+	}
+
+	/**
+	 * Returns true if the profiler should be displayed (not completed and not skipped).
+	 *
+	 * @return bool
+	 */
+	public function should_show_profiler() {
+		// @todo Remove this once we have a proper way to dismiss the profiler.
+		if ( ! defined( 'WOOCOMMERCE_ADMIN_DEV_SHOW_PROFILER' ) || false === WOOCOMMERCE_ADMIN_DEV_SHOW_PROFILER ) {
+			return false;
+		}
+		$onboarding_data = get_option( 'wc_onboarding_profile', array() );
+
+		// @todo Update this to compare to proper bools (https://github.com/woocommerce/woocommerce-admin/issues/2299).
+		$is_completed = isset( $onboarding_data['completed'] ) && 'true' === $onboarding_data['completed'];
+		$is_skipped   = isset( $onboarding_data['skipped'] ) && 'true' === $onboarding_data['skipped'];
+		return $is_completed || $is_skipped ? false : true;
```

Comment:
What happens if a user has already gone through the default WC setup wizard?  Do they get hit with this again or do we want to add a note to change this before merging into core?

Reply:
Good question. We should probably respect the current flag.

Reply:
There actually isn't a flag for this exactly. The redirect and notice both just rely on the presence of the `install` notice showing. 

https://github.com/woocommerce/woocommerce/blob/af574f6484a4300d1ec92c569593013f2e2fced2/includes/admin/class-wc-admin.php#L161-L164

https://github.com/woocommerce/woocommerce/blob/d0491072e8a7d0a4535db0ec2bae4a6d43ff64fd/includes/admin/views/html-notice-install.php

I think instead of relying on the old notice, we should probably add a note and/or an issue here on GitHub with the label 'On Core Merge'.

I'm thinking when we merge onboarding to core, we could set the completed flag to true during the upgrade process.

Reply:
Sounds like a great solution to me!